src/Models/Discussion.php

```diff
@@ -130,4 +132,17 @@ public function watchers(): BelongsToMany
     {
         return $this->belongsToMany(User::class);
     }
+
+    public function scopeWithIsBeingWatched(Builder $builder, $user)
+    {
+        $builder->when(config('firefly.features.watchers') && $user, function ($query) use ($user) {
+            $query->leftJoin('discussion_user', function (JoinClause $join) use ($user) {
+                $join->on('discussion_user.discussion_id', '=', 'discussions.id')
+                    ->where('discussion_user.user_id', $user->id);
+            })->select([
+                'discussions.*',
+                'discussion_user.user_id as is_being_watched',
+            ]);
+        });
```

Comment:
I found a much easier/cleaner way to do this! 😁 
```suggestion
        $builder->when(config('firefly.features.watchers'), function ($query) use ($user) {
            $query->withExists(['watchers as is_being_watched' => fn ($query) => $query->where('user_id', $user->id));
        });
```

Reply:
That's really pretty!
Composer requires `"php": "^7.2 || ^8.0",` and short functions are available starting in 7.4, is that an issue?

Reply:
In that case, let's update composer to require 7.4 at a minimum.

Reply:
Ok, I can make that part of my next commit and this PR.
Thanks

Reply:
This package has 7.4 as a minimum.

https://github.com/AndreasElia/laravel-firefly/blob/f55ce93d0dd5e705172b8e3380c734da4596960c/composer.json#L22

Reply:
Weird, sorry about that, my IDE must have pulled up the wrong composer.json file.
All good, will send the last updates shortly!